### DISCUSSION TITLE
One place for all types of telemetry events!!

### DISCUSSION THREAD

**Discussion by Abhign** [Sunbird-Ed/Community#304] [Documentation]
Can we get all types of system-generated telemetry events and all the page-wise telemetry events by building block-wise in one place?  So that it would ease the debugging and if any 3rd party adopts this platform they can send only the relevant events also it would help in generating the Adhoc scripts & new reports.

Here is the reference for  Knowledge building block that has been created  https://knowlg.sunbird.org/learn/product-and-developer-guide/player/telemetry-events

We are looking for the same kind of document from all the building blocks.

**Comment by Krishnaj20** [comment]
Thanks @Abhign for starting this thread. 
@swayangjit @rajeevsathish - We discussed about optimizing the telemetry events related information in the ED microsite. Can you take this thread forward?

@vinukumar-vs

**Reply by vinukumar-vs** [reply]
@rajeevsathish @swayangjit If we attach the Instrumentation sheets for Portal & Mobile with all the possible individual events documented in the microsite & attaching the instrumentation sheet for more reference should be good enough.

We should update the below page with more details & samples as documented by [Knowlg](https://knowlg.sunbird.org/learn/product-and-developer-guide/player/telemetry-events)
https://ed.sunbird.org/learn/product-and-developers-guide/observability#what-is-telemetry

**Reply by Krishnaj20** [reply]
@vinukumar-vs - the ask from @Abhign was to include both the points you mentioned in your previous reply.

**Reply by Krishnaj20** [reply]
@rhwarrier @reshmi-nair @pallakartheekreddy @luckynvdu @surendrasinghs - This was previously discussed as one of the priority asks from DIKSHA. Please plan to include this documentation in your respective BB microsites.

**Reply by chitranshu-keshav** [reply]
@vinukumar-vs @rajeevsathish @swayangjit Can you please update this thread with the completion status. If it is done then please get this thread marked as complete.
cc @Krishnaj20
